### PR TITLE
Fix the deadlock issue of Manager db under high concurrency

### LINKED ISSUE #2099: [Bug] [1.1.2]EngineConn may deadlock when creating and modifying label information under high concurrency
### Search before asking

- [X] I searched the [issues](https://github.com/apache/incubator-linkis/issues) and found no similar issues.

### Linkis Component

linkis-cg-manager

### What happened + What you expected to happen

Under high concurrency, engineConn may deadlock when creating and modifying label information, and the probability of occurrence is relatively low. This is due to the problem of batch modification of unique keys. It is recommended to optimize iterative modification in the program.

### Relevent platform

all

### Reproduction script

no

### Anything else

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by peacewong**
### What is the purpose of the change
close #2099

### Brief change log
- Fix Deadlock issue that update label keyValue

### CODE REVIEW

File: linkis-computation-governance/linkis-manager/linkis-manager-persistence/src/main/java/org/apache/linkis/manager/persistence/impl/DefaultLabelManagerPersistence.java [hunk 1]
```diff
@@ -113,15 +114,26 @@ public void removeLabel(PersistenceLabel persistenceLabel) {
...
+            replaceIntoLabelKeyValues(persistenceLabel.getValue(), id);
+        }
+    }
+
+    private void replaceIntoLabelKeyValues(
+            Map<String, String> labelValueKeyAndContent, int labelId) {
+        if (null != labelValueKeyAndContent && labelId > 0) {
+            Iterator<Map.Entry<String, String>> iterator =
+                    labelValueKeyAndContent.entrySet().iterator();
+            while (iterator.hasNext()) {
```
Comment (wForget) [inline linkis-computation-governance/linkis-manager/linkis-manager-persistence/src/main/java/org/apache/linkis/manager/persistence/impl/DefaultLabelManagerPersistence.java:130]: use foreach? like:
```
for(Map.Entry<String, String> labelKeyValue: labelValueKeyAndContent.entrySet())
```

Comment (peacewong) [reply linkis-computation-governance/linkis-manager/linkis-manager-persistence/src/main/java/org/apache/linkis/manager/persistence/impl/DefaultLabelManagerPersistence.java:130]: > use foreach? like:
> 
> ```
> for(Map.Entry<String, String> labelKeyValue: labelValueKeyAndContent.entrySet())
> ```

It is better to use iterator for map iteration, considering safety and performance

Comment (wForget) [reply linkis-computation-governance/linkis-manager/linkis-manager-persistence/src/main/java/org/apache/linkis/manager/persistence/impl/DefaultLabelManagerPersistence.java:130]: got it, thanks.

**Review by wForget — APPROVED** [review]
LGTM

**Pull request merged**